buttercup.el

```diff
@@ -1969,6 +1972,29 @@ With buttercup minor mode active the following is activated:
       (cl-dolist (form imenu-forms)
         (setq imenu-generic-expression (delete form imenu-generic-expression))))))
 
+(defun buttercup-error-when-loading-old-elc-files (elc-file)
+  "Raise an error when ELC-FILE is an elc-file and older than its el-file."
+  (when (string= (file-name-extension elc-file t) ".elc")
+    (let ((elfile (substring elc-file 0 -1))
+          elc-attributes el-attributes)
+      (when (file-exists-p elfile)
+        (setq elc-attributes (file-attributes elc-file))
+        (setq el-attributes (file-attributes elfile))
+        (if (time-less-p (file-attribute-modification-time elc-attributes)
```

Comment:
When do we use `when` vs `if`?

Reply:
We generally use when in buttercup, I think.  There is no hard rule.  I've recently used plain `and` and `or` when I felt it made the code easier to understand.